### ISSUE TITLE
OTA failure when using in parallel with WiFiClientSecure

### ISSUE THREAD

**Issue by NikuColesnic** [esp8266/Arduino#3158]
### Basic Infos
Around the end of March I updated the ESP8266 toolchain within the Arduino IDE (which we are using for developement for the ESP WROOM-02). And abruptly the OTA update part of the firmware started crashing.  @igrr I noticed that around that period of time you made commits regarding the axTLS implementation, which in turn fixed for us the connection problems we have had with  WiFiClientSecure (we are using MQTT to send data to the Bluemix IOTF platform). 

Long story short, I have the following weird situation:
1. Using WiFiClientSecure, with port 8883/443 and the OTA URL over HTTP(1.a.) or HTTPS(1.b.) crashes.
2. Using WiFiClient with port 1883 and the OTA URL over HTTPS(2.a) crashes, over HTTP works.

In all the cases it crashes at this function call : **ESPhttpUpdate.update()**

**What could be the problem for this failure ?**

*See debug messages below.

#### Hardware
Hardware:			Espressif WROOM-02

### Description

Problem description

### Settings in IDE

Module:  			Generic ESP8266 Module
Flash Size: 		2MB/1MB
CPU Frequency:		80Mhz
Flash Mode:			DIO
Flash Frequency:	40Mhz
Upload Using:	        SERIAL
Reset Method:		ck

### **Debug Messages**
++++
**1.a.**
++++
******************************************************************************************************
Exception (29):
epc1=0x4021329c epc2=0x00000000 epc3=0x40000f68 excvaddr=0x00000000 depc=0x00000000

ctx: cont 
sp: 3fff4870 end: 3fff4e50 offset: 01a0

>>>stack>>>
3fff4a10:  4025ab60 ffffc64b 00000000 40213293  
3fff4a20:  000678e0 00000000 3fff3d8c 3fff3d8c  
3fff4a30:  00000000 00000000 3fffae2c 402154f4  
3fff4a40:  000678e0 00000000 3fffae2c 4021374d  
3fff4a50:  00001000 000678e0 3fff4a90 40213a86  
3fff4a60:  3ffec3c0 00000000 3fff4ad0 00000001  
3fff4a70:  3fff3d8c 000678e0 3fff3234 4020ec13  
3fff4a80:  3ffeb670 00000000 000003e8 40213b68  
3fff4a90:  3fffb1b4 0000000f 00000000 4020e1a0  
3fff4aa0:  3fffae2c 3fff3d8c 3fff4ad0 3fff3e30  
3fff4ab0:  402127d4 001540c8 3fffae2c 000000c8  
3fff4ac0:  3fffae2c 3fff3234 3fff4b20 4020ef72  
3fff4ad0:  3fff6e0c 0000000f 00000000 3ffea9a0  
3fff4ae0:  00000000 00000000 00000aef 4010020c  
3fff4af0:  00000000 000678e0 00000000 4010068c  
3fff4b00:  00000000 00000000 00000000 3fff3230  
3fff4b10:  3fff4d54 3fff3234 3fff3234 4020f075  
3fff4b20:  3fff5e94 3fffae2c 3fffae94 0000001f  
3fff4b30:  00000010 3f000050 00011f40 3fffaebc  
3fff4b40:  0000002f 00000021 3fffad24 0000000f  
3fff4b50:  00000004 3fffaef4 0000013f 00000135  
3fff4b60:  3fffad9c 0000001f 00000013 3fffadc4  
3fff4b70:  0000000f 00000000 3fffacf8 00000001  
3fff4b80:  000000c8 000678e0 3fff4b00 00000000  
3fff4b90:  00000000 00000000 00000000 40213a86  
3fff4ba0:  3fff4d54 3fff4bc0 3fff3d68 40213078  
3fff4bb0:  3fff4d54 3fff3234 3fff3d68 4020d74b  
3fff4bc0:  3fff6e6c 0000000f 00000003 00000acb  
3fff4bd0:  3fff3d2c 00000106 00000106 3fff4c70  
3fff4be0:  00000020 3fff4c70 3fff4c70 402139bb  
3fff4bf0:  3fff5e6c 00000018 3fff4c70 40213a07  
3fff4c00:  4021551c 3fffc6fc 3fff4c70 40213a39  
3fff4c10:  00000018 3fff4ca0 3fff4c70 40213bad  
3fff4c20:  3fff5e6c 00000001 3fff4c70 40213a07  
3fff4c30:  3fff3d2c 00000acb 00000acb 4010020c  
3fff4c40:  00000001 3fff4cf0 3fff4c70 4010068c  
3fff4c50:  3ffeac94 3fff4cf0 3fff4ca0 40213960  
3fff4c60:  3fff3d2c 00000301 00000301 4010020c  
3fff4c70:  00000025 3fff4d30 3fff4cac 4010068c  
3fff4c80:  3ffeac50 3fff4d30 3fff4cfc 3fff3d2c  
3fff4c90:  00000030 3fff4d54 00000000 4010053d  
3fff4ca0:  3fff3d2c 000000fe 000000fe 4010020c  
3fff4cb0:  3fff3d2c 00000313 00000313 3fff2efc  
3fff4cc0:  00000030 3fff4d54 3fff4d54 402139bb  
3fff4cd0:  3fff982c 00000020 3fff4d54 40213a07  
3fff4ce0:  00000000 3fff4d30 3fff4d54 40213a39  
3fff4cf0:  3fff4dc0 3fff4d48 00000000 00000000  
3fff4d00:  00000001 3ffec3c0 3fff4d30 40213b68  
3fff4d10:  3ffe007d 3fff4de0 00000000 3fff2efc  
3fff4d20:  3fff4dc0 3ffec3c0 00000000 40209948  
3fff4d30:  3fff6efc 0000000f 00000002 3fff97f4  
3fff4d40:  0000001f 00000018 3fffacbc 0000002f  
3fff4d50:  00000028 3fffad54 0000003f 00000038  
3fff4d60:  00000004 3fff4d01 3fff97bc 0000002f  
3fff4d70:  00000026 3fff4de0 3fff2f38 4020bcb8  
3fff4d80:  3fff2f7f 00000003 3fff4da0 00000002  
3fff4d90:  0000003e 3fff2f38 00000041 3fff4de0  
3fff4da0:  3fff4de0 0001b5a7 3fff2f38 40215514  
3fff4db0:  3fff4de0 00000001 00000019 4020c35a  
3fff4dc0:  2d746f69 6d632f32 70752f64 65746164  
3fff4dd0:  746d662f 6f736a2f 3fff006e 4020c270  
3fff4de0:  3ffe9f01 3fff2f38 402145d8 3fff3e30  
3fff4df0:  402030ba 000003e8 000003e8 402030af  
3fff4e00:  3ffea002 3ffe9f70 3fff3d68 3fff3e24  
3fff4e10:  3fffdad0 00000000 00000001 402096d8  
3fff4e20:  3fffdad0 00000000 00000001 402091f2  
3fff4e30:  feefeffe 00000000 3fff3e1c 40214624  
3fff4e40:  feefeffe feefeffe 3fff3e30 4010070c  
<<<stack<<<

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v00000000
~ld

*************************************************************************************
++++
**1.b.**
++++

Exception (29):
epc1=0x4021d092 epc2=0x00000000 epc3=0x40235ebd excvaddr=0x00000000 depc=0x00000000

ctx: cont 
sp: 3fff47f0 end: 3fff4e50 offset: 01a0

>>>stack>>>
3fff4990:  00000000 00001ad0 3fffb264 401004f4  
3fff49a0:  00000001 3fff6c14 3fffb264 40248115  
3fff49b0:  3fff6c14 3fffac84 3fffb1bc 40249a17  
3fff49c0:  00000000 3fffaf34 3fffac84 401004d8  
3fff49d0:  4020b008 00005958 3fff3e30 3fff30dc  
3fff49e0:  3fffac84 3fffaf34 3fffb1bc 4020b6c6  
3fff49f0:  000001bb 3fffb104 3fffaf34 4020a7e6  
3fff4a00:  038fc068 00000000 3fff4a40 00000000  
3fff4a10:  000001bb 3fffaf34 3fffac84 4020ba1d  
3fff4a20:  3ffeb630 038fc068 3ffeb630 038fc068  
3fff4a30:  3fff4bc0 00000000 3fff4b10 4020e29c  
3fff4a40:  00000000 00000000 3fffaf0c 40213a07  
3fff4a50:  00000000 00000000 3fff4b10 4020e77c  
3fff4a60:  3fffaf18 00000000 3fffaf00 40213bad  
3fff4a70:  3ffea9a0 00000000 00000000 40214428  
3fff4a80:  3ffea928 00000b13 3fff4b10 4020e14b  
3fff4a90:  3fff4acc 00000000 3fff4ac0 3fff3d48  
3fff4aa0:  3fff4bc0 3fff3234 3fff4b10 4020e82a  
3fff4ab0:  3fff4bc0 3fff3234 3fff4b10 4020ee91  
3fff4ac0:  00000000 00000000 00000000 3ffea9a0  
3fff4ad0:  00000000 00000000 00000af6 4010020c  
3fff4ae0:  00000000 3fff3234 3fff4b8c 4010068c  
3fff4af0:  3fff4ba0 00000023 3fff4bc0 3fff4d48  
3fff4b00:  3fff4d54 3fff3234 3fff4bc0 4020f0e1  
3fff4b10:  3fffac9c 3fffaf34 3fffac84 0000000f  
3fff4b20:  0000000d 000001bb 00011f40 3fffaf54  
3fff4b30:  0000005f 00000057 3fffada4 0000000f  
3fff4b40:  00000005 3fffafbc 0000013f 00000135  
3fff4b50:  3fffadd4 0000001f 00000013 3fffadfc  
3fff4b60:  0000000f 00000000 3fffaf00 00000001  
3fff4b70:  00000000 ffffffff 3fff4b00 00000000  
3fff4b80:  00000000 00000000 00000000 00000000  
3fff4b90:  00000000 00000000 3fff4bc0 40213a86  
3fff4ba0:  3ffea7c8 00000316 3fff3d68 3fff3230  
3fff4bb0:  3fff4d54 3fff3234 3fff3d68 4020d730  
3fff4bc0:  3fff5e7c 0000000f 00000003 00000311  
3fff4bd0:  3fff3d2c 00000839 00000839 3fff4c70  
3fff4be0:  00000020 3fff4c70 3fff4c70 402139bb  
3fff4bf0:  3fff9804 00000018 3fff4c70 40213a07  
3fff4c00:  39c246ed 3aa76a6e 3fff4c70 40213a39  
3fff4c10:  00000018 3fff4ca0 3fff4c70 40213bad  
3fff4c20:  3fff9804 00000001 3fff4c70 40213a07  
3fff4c30:  3fff3d2c 00000311 00000311 4010020c  
3fff4c40:  00000001 3fff4cf0 3fff4c70 4010068c  
3fff4c50:  3ffeac94 3fff4cf0 3fff4ca0 40213960  
3fff4c60:  3fff3d2c 00000104 00000104 4010020c  
3fff4c70:  00000025 3fff4d30 3fff4cac 4010068c  
3fff4c80:  3ffeac50 3fff4d30 3fff4cfc 3fff3d2c  
3fff4c90:  00000060 3fff4d54 00000000 4010053d  
3fff4ca0:  3fff3d2c 000000fe 000000fe 4010020c  
3fff4cb0:  3fff3d2c 00000101 00000101 3fff2efc  
3fff4cc0:  00000060 3fff4d54 3fff4d54 402139bb  
3fff4cd0:  3fffacc4 00000054 3fff4d54 40213a07  
3fff4ce0:  00000000 3fff4d30 3fff4d54 40213a39  
3fff4cf0:  3fff4dc0 3fff4d48 00000001 00000001  
3fff4d00:  00000001 3ffec3c0 3fff4d30 40213b68  
3fff4d10:  3ffe007d 3fff4de0 00000000 3fff2efc  
3fff4d20:  3fff4dc0 3ffec3c0 00000000 40209948  
3fff4d30:  3fff6eec 0000000f 00000002 3fff5e54  
3fff4d40:  0000001f 00000018 3fffac4c 0000002f  
3fff4d50:  00000028 3fffad2c 0000006f 0000006c  
3fff4d60:  00000004 3fff4d01 3fff9784 0000002f  
3fff4d70:  00000026 3fff4de0 3fff2f38 4020bcb8  
3fff4d80:  3fff2f7f 00000003 3fff4da0 00000002  
3fff4d90:  0000003e 3fff2f38 00000041 3fff4de0  
3fff4da0:  3fff4de0 0000fdfc 3fff2f38 40215514  
3fff4db0:  3fff4de0 00000001 00000019 4020c35a  
3fff4dc0:  2d746f69 6d632f32 70752f64 65746164  
3fff4dd0:  746d662f 6f736a2f 3fff006e 4020c270  
3fff4de0:  3ffe9f01 3fff2f38 402145d8 3fff3e30  
3fff4df0:  402030ba 000003e8 000003e8 402030af  
3fff4e00:  3ffea002 3ffe9f70 3fff3d68 3fff3e24  
3fff4e10:  3fffdad0 00000000 00000001 402096d8  
3fff4e20:  3fffdad0 00000000 00000001 402091f2  
3fff4e30:  feefeffe 00000000 3fff3e1c 40214624  
3fff4e40:  feefeffe feefeffe 3fff3e30 4010070c  
<<<stack<<<

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v00000000
~ld

******************************************************************************************************
++++
**2.a.**
++++
Exception (29):
epc1=0x4021d092 epc2=0x00000000 epc3=0x4022409f excvaddr=0x00000100 depc=0x00000000

ctx: cont 
sp: 3fff4650 end: 3fff4e50 offset: 01a0

>>>stack>>>
3fff47f0:  00800000 00000080 3fff7234 4024b65a  
3fff4800:  ffffffff 3fff9d5c 3fff7234 4024b7ec  
3fff4810:  00000040 3fff9d5c 3fffb08c 4024bbd5  
3fff4820:  3fffbfc4 3fff9d5c 3fffb08c 4024c5c1  
3fff4830:  3fff9dfc aec5c182 4ff7dd9c 00000001  
3fff4840:  3fff9d5c 3fffb074 3fffb08c 00000001  
3fff4850:  00000010 3fff9d5c 0000000f 4024c830  
3fff4860:  3fff7234 3fffa03c 0000000f 00000001  
3fff4870:  00000001 3fffa054 3fffa054 4024b6bc  
3fff4880:  00000100 3fff9dcc 3fff797f 00000000  
3fff4890:  00000100 3fff9dcc 3fff797f 4024d951  
3fff48a0:  3fff48d0 00000000 000000d0 00000030  
3fff48b0:  747be88a 3fff7398 3fff72e8 3fff7979  
3fff48c0:  00000004 3fff789c 3fff7979 40249886  
3fff48d0:  139f0303 822ffd1e ba2de9f8 9461d6f2  
3fff48e0:  33a94686 04c5d7da f1a4a423 cf079e78  
3fff48f0:  b5b81742 a2863beb c61baa10 e88a8c97  
3fff4900:  00000004 00000004 3fff789c 40249c71  
3fff4910:  3fff7234 dd149adb 0872a036 8f202977  
3fff4920:  69f6e6a5 600011f0 3fff77e4 3fff755c  
3fff4930:  00000000 00000004 00000004 4020b278  
3fff4940:  0000000e 00000bce 3fff9464 40214650  
3fff4950:  0000004a 00000000 0000000d 3fff7979  
3fff4960:  3fff7979 00000004 3fff789c 3fff7979  
3fff4970:  00000004 00000004 3fff789c 40249672  
3fff4980:  00000043 00000016 3fff789c 402087a1  
3fff4990:  5009b18f 3fff752c 00000000 00000000  
3fff49a0:  3fff49f0 0000000e 00000010 402499bb  
3fff49b0:  3fff752c 3fff7454 3fff789c 01000000  
3fff49c0:  3fff49f0 3fff7614 3fff789c 402497d8  
3fff49d0:  4020b008 00000000 3fff758c 00001387  
3fff49e0:  0000e850 3fff7614 3fff758c 4020b705  
3fff49f0:  000001bb 3fff77e4 3fff7614 4020a7e6  
3fff4a00:  038fc068 00000000 3fff4a40 00000000  
3fff4a10:  000001bb 3fff7614 3fff7454 4020ba1d  
3fff4a20:  3ffeb630 038fc068 3ffeb630 038fc068  
3fff4a30:  3fff4bc0 00000000 3fff4b10 4020e29c  
3fff4a40:  00000000 00000000 3fff75ec 40213a07  
3fff4a50:  00000000 00000000 3fff4b10 4020e77c  
3fff4a60:  3fff75f8 00000000 3fff75e0 40213bad  
3fff4a70:  3ffea9a0 00000000 00000000 40214428  
3fff4a80:  3ffea928 000003ef 3fff4b10 4020e14b  
3fff4a90:  3fff4acc 00000000 3fff4ac0 3fff3d40  
3fff4aa0:  3fff4bc0 3fff322c 3fff4b10 4020e82a  
3fff4ab0:  3fff4bc0 3fff322c 3fff4b10 4020ee91  
3fff4ac0:  00000000 00000000 00000000 3ffea9a0  
3fff4ad0:  00000000 00000000 000003d2 4010020c  
3fff4ae0:  00000000 3fff322c 3fff4b8c 4010068c  
3fff4af0:  3fff4ba0 00000023 3fff4bc0 3fff4d48  
3fff4b00:  3fff4d54 3fff322c 3fff4bc0 4020f0e1  
3fff4b10:  3fff746c 3fff7614 3fff7454 0000000f  
3fff4b20:  0000000d 000001bb 00011f40 3fff7634  
3fff4b30:  0000005f 00000057 3fff7484 0000000f  
3fff4b40:  00000005 3fff769c 0000013f 00000135  
3fff4b50:  3fff74b4 0000001f 00000013 3fff74dc  
3fff4b60:  0000000f 00000000 3fff75e0 00000001  
3fff4b70:  00000000 ffffffff 3fff4b00 00000000  
3fff4b80:  00000000 00000000 00000000 00000000  
3fff4b90:  00000000 00000000 3fff4bc0 40213a86  
3fff4ba0:  3ffea7c8 0000038f 3fff3d60 3fff3228  
3fff4bb0:  3fff4d54 3fff322c 3fff3d60 4020d730  
3fff4bc0:  3fff72cc 0000000f 00000003 0000038a  
3fff4bd0:  3fff3d24 00000106 00000106 3fff4c70  
3fff4be0:  00000020 3fff4c70 3fff4c70 402139bb  
3fff4bf0:  3fff5e6c 00000018 3fff4c70 40213a07  
3fff4c00:  40101bd9 3fff05a0 3fff4c70 40213a39  
3fff4c10:  00000018 3fff4ca0 3fff4c70 40213bad  
3fff4c20:  3fff5e6c 00000001 3fff4c70 40213a07  
3fff4c30:  3fff3d24 0000038a 0000038a 4010020c  
3fff4c40:  00000001 3fff4cf0 3fff4c70 4010068c  
3fff4c50:  3ffeac94 3fff4cf0 3fff4ca0 40213960  
3fff4c60:  3fff3d24 00000016 00000016 4010020c  
3fff4c70:  00000025 3fff4d30 3fff4cac 4010068c  
3fff4c80:  3ffeac50 3fff4d30 3fff4cfc 3fff3d24  
3fff4c90:  00000060 3fff4d54 00000000 4010053d  
3fff4ca0:  3fff3d24 000000fe 000000fe 4010020c  
3fff4cb0:  3fff3d24 00000013 00000013 3fff2efc  
3fff4cc0:  00000060 3fff4d54 3fff4d54 402139bb  
3fff4cd0:  3fff72f4 00000054 3fff4d54 40213a07  
3fff4ce0:  00000000 3fff4d30 3fff4d54 40213a39  
3fff4cf0:  3fff4dc0 3fff4d48 00000001 00000001  
3fff4d00:  00000001 3ffec3c0 3fff4d30 40213b68  
3fff4d10:  3ffe007d 00000000 00000000 3fff2efc  
3fff4d20:  3fff4dc0 3ffec3c0 00000000 40209948  
3fff4d30:  3fff5e54 0000000f 00000002 3fff56fc  
3fff4d40:  0000001f 00000018 3fff727c 0000002f  
3fff4d50:  00000028 3fff73dc 0000006f 0000006c  
3fff4d60:  00000004 3fff4d01 3fff735c 0000002f  
3fff4d70:  00000026 3fffdad0 3fff2f38 4020bcb8  
3fff4d80:  3fff2f7f 00000003 3fff4da0 00000002  
3fff4d90:  0000003e 3fff2f38 00000041 3fff4de0  
3fff4da0:  3fff4de0 0000e6d2 3fff2f38 40215514  
3fff4db0:  3fff4de0 00000001 00000019 4020c35a  
3fff4dc0:  2d746f69 6d632f32 70752f64 65746164  
3fff4dd0:  746d662f 6f736a2f 3fff006e 4020c270  
3fff4de0:  3ffe9f01 3fff2f38 402145d8 3fff3e30  
3fff4df0:  402030ba 000003e8 000003e8 402030af  
3fff4e00:  3ffea002 3ffe9f70 3fff3d60 3fff3e1c  
3fff4e10:  3fffdad0 00000000 00000001 402096d8  
3fff4e20:  3fffdad0 00000000 00000001 402091f2  
3fff4e30:  feefeffe 00000000 3fff3e14 40214624  
3fff4e40:  feefeffe feefeffe 3fff3e30 4010070c  
<<<stack<<<

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v00000000
~ld

****************************************************************************************************

**Comment by earlephilhower** [comment]
Closing as the stack traces are not decoded, it involved AxTLS, and there being no MCVE.  If you're still have an issue with the 2.5.2 release or git head, please do file a new issue with the required info filled out so we can have a look at it.
